### PR TITLE
hw1 completion 

### PR DESCRIPTION
**Pull request by imOnVacation**
The hw1 contains two exercises

1. Flexbox Froggy game 
The exercise was just playing game with css knowledges, fun exercise! 

2. Interactive small function to check if user input number is palindrome or not. 
The main goal is to check whether the user's input number is palindrome, and styling the HTML page. I added two elements to the HTML pages to display error message / result. They will pops out regarding to different input. For example, if the user enters an invalid number, the input box border will turn red, and an error message will appear. If the input is valid, a result message will display whether the number is a palindrome or not.

- UI 
WeChat9a990c8c0a8e9229d52e636ba0fa5fb6

- Invalid input 
WechatIMG565

- Valid input but is not palindrome 
image

- Valid input but is palindrome 
image

### CODE REVIEW

File: hw1/02-palindrome.js [hunk 1]
```diff
@@ -1,3 +1,45 @@
 const elem = document.querySelector('input');
+const errorElem = document.querySelector('.inputError');
+const resultElem = document.querySelector('.displayResult');
 
 elem.addEventListener('input', handleInput);
```
Comment (colonelcandycorn) [inline hw1/02-palindrome.js]: Airbnb Linter gave this message:
   5:32  error  'handleInput' was used before it was defined   no-use-before-define

File: hw1/02-palindrome.js [hunk 2]
```diff
@@ -1,3 +1,45 @@
...
+    errorElem.textContent = 'Enter a valid positive number.';
+    resultElem.textContent = '';
+    return;
+  }
+
+  // Reset error and check for palindrome
+  errorElem.textContent = '';
+  elem.style.borderColor = '';
+
+  if (isPalindrome(inputValue)) {
```
Comment (colonelcandycorn) [inline hw1/02-palindrome.js:35]: Linter:
 32:9   error  'isPalindrome' was used before it was defined  no-use-before-define

**Review by colonelcandycorn — APPROVED** [review]
Everything works perfect. I set up my .eslintrc to run airbnb linter so I just ran it on your code and added the comments it gave.

File: hw1/02-palindrome.js [hunk 3]
```diff
@@ -1,3 +1,45 @@
 const elem = document.querySelector('input');
+const errorElem = document.querySelector('.inputError');
+const resultElem = document.querySelector('.displayResult');
 
 elem.addEventListener('input', handleInput);
+
+function handleInput() {
+  const inputValue = elem.value;
+  const numValue = Number(inputValue);
```
Comment (Ghaadah) [inline hw1/02-palindrome.js]: I believe the numValue function is unnecessary. The input type in the HTML is set to "number" which ensures that InputValue will be always a number. therefore, converting it to a number is not needed.

File: hw1/02-palindrome.js [hunk 4]
```diff
@@ -1,3 +1,45 @@
...
+    resultElem.textContent = '';
+    errorElem.textContent = '';
+    elem.style.borderColor = '';
+    return;
+  }
+
+  // Check if the input is a negative number
+  if (numValue < 0) {
+    errorElem.style.color = 'red';
+    elem.style.borderColor = 'red';
```
Comment (Ghaadah) [inline hw1/02-palindrome.js:25]: I  like that you used a border for the error msg! I noticed that you've separated the errorElem and resultElem , which gives you control to apply different styles to each.

**Review by Ghaadah — COMMENTED** [review]
Great job in both exercises. I've included some feedback for your review.

File: hw1/02-palindrome.js (hunk 3 above)
Comment (imOnVacation) [reply hw1/02-palindrome.js]: You're right! Just fixed it, thanks!

File: hw1/02-palindrome.js (hunk 2 above)
Comment (imOnVacation) [reply hw1/02-palindrome.js:35]: Thanks Sarah! I updated the code accordingly to your feedback.

File: hw1/02-palindrome.js [hunk 5]
```diff
@@ -1,3 +1,44 @@
...
+  const inputValue = elem.value;
+
+  // Clear the error and border when input is empty
+  if (!inputValue) {
+    resultElem.textContent = '';
+    errorElem.textContent = '';
+    elem.style.borderColor = '';
+    return;
+  }
+
```
Comment (ruspedpdx) [inline hw1/02-palindrome.js:21]: I like that you clean up when there is nothing to see, nice touch!

**Review by ruspedpdx — COMMENTED** [review]
Well done, thank you

**Pull request merged**